lambda/tmc.ml

```diff
@@ -952,21 +952,22 @@ and traverse_binding :
   fun binding_kind outer_ctx inner_ctx binding ->
   let (var, def) : Ident.t * lambda =
     match binding_kind, binding with
-    | Recursive, { id; rkind=_; def } -> id, def
+    | Recursive, { id; def } -> id, Lfunction def
     | Non_recursive, (var, def) -> var, def
   in
-  let mk_same_binding (var : Ident.t) (def : lambda) : a =
-    match binding_kind, binding with
-    | Recursive, { id=_; rkind; def=_ } -> { id = var; rkind; def }
-    | Non_recursive, _ -> var, def
-  in
-  let mk_static_binding (var : Ident.t) (def : lambda) : a =
-    match binding_kind, binding with
-    | Recursive, _ -> { id = var; rkind = Static; def }
-    | Non_recursive, _ -> var, def
+  let mk_binding (var : Ident.t) (def : lambda) : a =
+    match binding_kind with
+    | Recursive ->
+        let def =
+          match def with
+          | Lfunction lfun -> lfun
+          | _ -> assert false
```

Comment:
I wonder if we could avoid this not-so-nice code by adding yet another parameter to `binding_kind`, to decide whether the internal term is a `lambda` or a `lfunction`.